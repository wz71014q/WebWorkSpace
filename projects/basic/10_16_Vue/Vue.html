<!doctype html>
<html>
<head>
<meta charset="utf-8">
<title>Vue</title>
</head>

<body>
<div id="app">{{message}}</div>

<script type="text/javascript" src="vue.js"></script>

<script type="text/javascript" src="Vue_act.js"></script>
</body>
</html>
