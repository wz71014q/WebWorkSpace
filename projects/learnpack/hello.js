function hello() {
  console.log('用webpack压缩需要输入：webpack --mode=development(或production, 默认是production) hello.js -o bundle.js');
}
hello();