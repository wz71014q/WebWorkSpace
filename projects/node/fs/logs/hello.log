hello~
文件系统~
