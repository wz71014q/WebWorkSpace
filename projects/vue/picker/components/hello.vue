<template>
  <div>
    hello {{ msg }}
  </div>
</template>

<script>
export default {
  name: 'Hello',
  data() {
    return {
      msg: 'world'
    }
  }
}
</script>