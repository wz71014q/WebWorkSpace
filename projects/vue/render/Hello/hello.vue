<template>
  <div>
    hello {{ msg }}
    <p class="text">I'm jest demo</p>
    <button class="count" @click="increment">Increment {{ count }}</button>
    <slot></slot>
  </div>
</template>

<script>

export default {
  name: 'Hello',
  data() {
    return {
      msg: 'picker',
      count: 0
    };
  },
  methods: {
    increment() {
      this.count++;
    }
  }
};
</script>
