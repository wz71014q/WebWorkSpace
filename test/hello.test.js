
import Vue from 'vue';
import sinon from 'sinon';
import { mount } from '@vue/test-utils';
import { renderToString } from '@vue/server-test-utils';
import hello from './testDemo/hello.vue';

let wrapper;
let vm;

describe('vue组件测试', () => {
  beforeEach(() => { // 每次测试前确保我们的测试实例都是是干净完整的。返回一个wrapper对象
    wrapper = mount(hello);
  });
  afterEach(() => {
    vm && vm.$destroy();
    wrapper && wrapper.destroy();
  }, 100);
  it('renders a div', () => {
    expect(wrapper.contains('div')).toBe(true);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('p标签的内容是I\'m jest demo', () => {
    expect(wrapper.find('.text').text())
      .toEqual('I\'m jest demo');
  });
  it('计数器', () => {
    const button = wrapper.find('.count');
    button.trigger('click');
    expect(button.text())
      .toEqual('Increment a 1');
    console.log(wrapper.vm.$data.count);
    button.trigger('click');
    expect(button.text())
      .toEqual('Increment a 2');
    console.log(wrapper.vm.$data.count);
  });
  it('querySelector', () => {
    const Constructor = Vue.extend(hello);
    vm = new Constructor().$mount();
    // console.log(vm.$el.querySelector('.text').textContent);
    // console.log(vm.$el.querySelector('.text').innerHTML);
    expect(vm.$el.querySelector('.text').innerHTML).toEqual('I\'m jest demo');
  });
  it('renderToString render component as a html', async () => {
    const str = await renderToString(hello);
    expect(str).toContain('<p class="text">I\'m jest demo</p>');
  });
  // it('spy', () => {
  //   const Constructor = Vue.extend(hello);
  //   vm = new Constructor().$mount();
  //   var incrementSpy = sinon.spy(vm, 'increment');
  //   vm.increment();
  //   console.log('incrementSpy.callCount' + incrementSpy.callCount);
  //   expect(incrementSpy).toHaveBeenCalled();
  //   incrementSpy.restore();
  // });
});
