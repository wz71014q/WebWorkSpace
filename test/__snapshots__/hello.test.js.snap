// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vue组件测试 renders a div 1`] = `
<div>
  
  hello picker
  
  <p
    class="text"
  >
    I'm jest demo
  </p>
   
  <button
    class="count"
  >
    Increment a 0
  </button>
   
</div>
`;
